Userland/Libraries/LibDeviceTree/DeviceTree.h

```diff
@@ -18,6 +19,67 @@
 
 namespace DeviceTree {
 
+// Devicetree Specification 0.4 (DTSpec): https://github.com/devicetree-org/devicetree-specification/releases/download/v0.4/devicetree-specification-v0.4.pdf
+
+class DeviceTree;
+class Node;
+
+class Address {
+public:
+    Address() = default;
+    Address(ReadonlyBytes data)
+        : m_raw(static_cast<decltype(m_raw)>(data))
+    {
+    }
+
+    ReadonlyBytes raw() const { return m_raw; }
+
+    static Address from_flatptr(FlatPtr flatptr)
+    {
+        BigEndian<FlatPtr> big_endian_flatptr { flatptr };
+
+        Address address;
+        address.m_raw.resize(sizeof(FlatPtr));
+        __builtin_memcpy(address.m_raw.data(), &big_endian_flatptr, sizeof(big_endian_flatptr));
+        return address;
+    }
+
+    ErrorOr<FlatPtr> as_flatptr() const
+    {
+        if (m_raw.size() == sizeof(u32))
+            return *bit_cast<BigEndian<u32>*>(m_raw.data());
+        if (m_raw.size() == 2 * sizeof(u32))
```

Comment:
Why not `sizeof(u64)`?

Reply:
Everything in a device tree is organised in 32bit cells, and this is 2 cells big,
I think other places in the device tree code do this too